multigrid.tex

```diff
@@ -1 +1,93 @@
-\subsection{\label{sec:multigrid}Multigrid}.
\ No newline at end of file
+\subsection{\label{sec:multigrid}Multigrid}
+The implementation of the multigrid solver algorithm in LFRic has taken a
+long time to develop for several reasons. Firstly, the new solver API,
+which enables easy configuration of different solver algorithms and
+pre-conditioners\footnote{Even nested solvers as currently deployed.},
+uses advanced Fortran OO abstract types. This has caused most
+compilers to have problems and significantly delayed the
+implementation. Secondly, implementing multigrid itself required
+several infrastructure changes. Multiple meshes, maps between these
+meshes and support for intergrid kernels in the PSyKAl API and in
+PSyclone. These have all been successfully implemented. Thirdly, when
+the multigrid algorithm itself was implemented the solver had problems
+converging.
+This was ultimately resolved as an issue with the vertical
+pre-conditioner and not multigrid itself. 
+
+The multigrid algorithm implemented here is a four-level V-cycle of
+the horizontal mesh, and used to pre-condition the Helmholtz or
+pressure solver. In fact, the pre-conditioned pressure field is taken
+as the ``solved'' outcome and no iterative, Krylov solver is applied
```

Comment:
"no iterative" -> "non-iterative"?

Reply:
Aah. It is no iterative solver, buy I guess my comma and Krylov rather muddies the sense of that sentence.
Revised to a simpler "no Krylov solver"